x-pack/solutions/observability/plugins/observability_onboarding/ui_tests/tests/custom_logs/configuration.spec.ts

```diff
@@ -0,0 +1,257 @@
+    await browserAuth.loginAsAdmin();
+    await customLogsPage.goto();
+  });
+
+  test('When user clicks the back button user goes to the onboarding home page', async ({
+    pageObjects: { customLogsPage },
+    page,
+  }) => {
+    await customLogsPage.clickBackButton();
+    expect(page.url()).toContain('/app/observabilityOnboarding');
+  });
+
+  test(`Users shouldn't be able to continue if logFilePaths is empty`, async ({
+    pageObjects: { customLogsPage },
+  }) => {
+    await expect(customLogsPage.logFilePathInput(0)).toHaveValue('');
+    await expect(customLogsPage.continueButton()).toBeDisabled();
+  });
+
+  test(`Users should be able to continue if logFilePaths is not empty`, async ({
+    pageObjects: { customLogsPage },
+  }) => {
+    await customLogsPage.logFilePathInput(0).fill('some/path');
+
+    await expect(customLogsPage.continueButton()).not.toBeDisabled();
+  });
+
+  test('Users can add multiple logFilePaths', async ({ pageObjects: { customLogsPage } }) => {
+    await customLogsPage.addLogFilePathButton().click();
+    await expect(customLogsPage.logFilePathInput(0)).toBeVisible();
+    await expect(customLogsPage.logFilePathInput(1)).toBeVisible();
+  });
+
+  test('Users can delete logFilePaths', async ({ pageObjects: { customLogsPage } }) => {
+    await customLogsPage.addLogFilePathButton().click();
+    await expect(customLogsPage.logFilePathList()).toHaveCount(2);
+
+    await customLogsPage.logFilePathDeleteButton(1).click();
+    await expect(customLogsPage.logFilePathList()).toHaveCount(1);
+  });
+
+  test('Dataset Name and Integration Name are auto generated if it is the first path', async ({
+    pageObjects: { customLogsPage },
+  }) => {
+    await customLogsPage.logFilePathInput(0).fill('myLogs.log');
+
+    await expect(customLogsPage.integrationNameInput()).toHaveValue('mylogs');
+    await expect(customLogsPage.datasetNameInput()).toHaveValue('mylogs');
+  });
+
+  test('Dataset Name and Integration Name are not generated if it is not the first path', async ({
+    pageObjects: { customLogsPage },
+  }) => {
+    await customLogsPage.addLogFilePathButton().click();
+    await customLogsPage.logFilePathInput(1).fill('myLogs.log');
+
+    await expect(customLogsPage.integrationNameInput()).toHaveValue('');
+    await expect(customLogsPage.datasetNameInput()).toHaveValue('');
+  });
+
+  test('Service name input should be optional allowing user to continue if it is empty', async ({
+    pageObjects: { customLogsPage },
+  }) => {
+    await customLogsPage.logFilePathInput(0).fill('myLogs.log');
+
+    await expect(customLogsPage.serviceNameInput()).toHaveValue('');
+    await expect(customLogsPage.continueButton()).not.toBeDisabled();
+  });
+
+  test('Users should expand and collapse the Advanced settings section', async ({
+    pageObjects: { customLogsPage },
+  }) => {
+    await expect(customLogsPage.namespaceInput()).not.toBeInViewport();
```

Comment:
Is using `toBeInViewport` here intentional or is it a better use case for `toBeVisible `?

Reply:
yeah, it was a workaround for the way EUI accordion works, but not a very good one. I've changed this test to check  the visibility of the whole enclosing container instead of individual input fields, this was I can use `toBeVisible()`